docs/routes.md

```diff
@@ -0,0 +1,268 @@
+
+* the default exported function (run on `GET`) does not automatically receive data from the database and must fetch it manually
+* the `put()` should return a database operation or array of operations (which may be wrapped in promises), rather than the component data itself
+* these files are _only_ run server-side, and will not be optimized to work in kiln
+
+```js
+module.exports = function (ref, locals) {
+  //return Promise with data
+  return Promise.resolve({"hey": "hey"});
+};
+module.exports.put = function (ref, data) {
+  //return Promise with operations to be performed in a batch
+  return Promise.resolve([{
+    type: 'put',
+    key: '/components/text',
+    value:'{"hey": "hey"}'}
+  ]);
+};
+```
+
+## Pages
+
+### Overview
+```
+/pages
+/pages/:name
+/pages/:name.html
+/pages/:name.json
+/pages/:name@:version
+/pages/:name@:version.html
+/pages/:name@:version.json
+```
+
+Pages consist of a layout and a list of areas.  Each area defined in an page maps to a area in the layout template.  For example:
+```json
+{
+  "layout": "/components/feature-layout",
+  "center": ["/components/article/instances/3vf3"],
+  "side": [
+    "/components/share",
+    "/components/newsletter"
+  ]
+}
+```
+The layout component in this example has two areas: center and side.  When the page is displayed, the components listed here will be placed into the matching areas in the layout.
+
+### Page Data
+
+GETs and PUTs to pages work similarly to components. API calls without an extension will update/return data for that page, while the `.json` and `.html` extensions will return the composed (page and layout, and their child components) JSON and HTML, respectively.
+
+### Publishing
+
+Publishing a page has a special convenience behavior where all components referenced by the page will also be published.  (Example: https://github.com/nymag/amphora/pull/78)
+
+## URIs
+
+```
+/uris/<base64(:path)>
+```
+
+A URI is used to redirect some slug or URI to another page or component.  They can also redirect to other uris (establishing a 301 redirect), or several uris can point to the same resource.  URIs are stored as Base64, so:
+
+- `example.com` is `/uris/ZXhhbXBsZS5jb20=` => `/pages/jdskla@published`
+- `example.com/other/` is `/uris/ZXhhbXBsZS5jb20vb3RoZXI=` => `/pages/4revd3s@published`
+
+A URI is assumed to be pointing at the `@published` version if another version is not provided.  Therefore, only published content or specially tagged versions can be publicly exposed through URIs.
+
+## Lists
+
+```
+/lists
+/lists/:id
+```
+
+Lists are a _temporary_ solution until search functionality is discussed.  It is currently used for the tags component and the autocomplete behavior as a temporary place to store lists of information.  It can store any list of data on a `PUT`, and return back the same list of data with a `GET`.
```

Comment:
```suggestion
Lists are a _temporary_ solution until search functionality is discussed. It is currently used for the tags component and the autocomplete behavior as a temporary place to store lists of information. It can store any list of data on a `PUT`, and return back the same list of data with a `GET`.
```